config/v1/types_infrastructure.go

```diff
@@ -189,6 +192,45 @@ const (
 	IBMCloudProviderTypeUPI IBMCloudProviderType = "UPI"
 )
 
+type CloudControllerManagerState string
+
+const (
+	// Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling that kubelets and other CCM consumers should use --cloud-provider=external.
+	CloudControllerManagerExternal CloudControllerManagerState = "External"
+
+	// Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling that kubelets and other CCM consumers should not set --cloud-provider flag.
+	CloudControllerManagerNone CloudControllerManagerState = "None"
+)
+
+type CloudControllerManagerSettings struct {
+	// State determines whether or not an external Cloud Controller Manager is expected to
```

Comment:
Godoc for fields should start with the json tag name rather than the Go field name.